lib/api/config.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+// Client configuration required to instantiate a Calico client interface.
+type ClientConfig struct {
```

Comment:
Probably want to stick some documentation on this struct as well. e.g which of these options are supported together? When are certain options mandatory?  What format should each field take?